crates/runtime/src/tools/builtin/table_schema.rs

```diff
@@ -97,10 +97,19 @@ impl SpiceModelTool for TableSchemaTool {
                 .await
                 .boxed()?;
 
-            let v = serde_json::value::to_value(schema).boxed()?;
+            let schema_value = serde_json::value::to_value(schema).boxed()?;
 
-            table_schemas.push(v);
+            let table_schema = serde_json::json!({
+                "table": t,
+                "schema": schema_value
+            });
```

Comment:
We weren't including the table name with the schema information returned, so it was probably confusing the model when multiple table schemas were returned previously.